handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -236,56 +236,66 @@ protected void deallocate() {
         super(peerHost, peerPort);
         OpenSsl.ensureAvailability();
         leak = leakDetection ? leakDetector.track(this) : null;
```

Comment:
can we just create this leak detector as the last line in the constructor (with a comment for justification)? This will also minimize the "risk" for letting `this` escaping the constructor and using unitized fields of this object (even though practically it shouldn't). The member variable is `final` so we shouldn't have to worry about re-ordering with instructions outside the scope of the constructor.